DM-11547: dockerize apache https proxy.

loitly opened the pull request:
- use letsencrypt for public site, and self-signed for development
- add support for authentication via mod_auth_openidc
- expand build script to exposes configurable global variables to javascript

**self-signed https proxy locally.**
cd ./firefly; docker/proxy-dev/proxyctl.sh update
this will create and run the apache proxy in docker.  access it on port 80 or 443.  i.e.  http://localhost/

**javascript global variables**
to make any configuration property a javascript global variable, append '__$' to it.  i.e.  __$help.base.url

robyww commented:
Loi and I discussed better ways to do global inserted variables.  It would work by building a global objects that includes such things as MODULE_NAME, SCRIPT_NAME, help.base.url and build one object and insert only that as global with webpack.  We think this would work. I created a separate ticket DM-11856.

robyww commented on docker/proxy/Dockerfile:
```diff
@@ -0,0 +1,17 @@
+#RUN apt-get update && \
+#    apt-get -f stretch install libapache2-mod-auth-openidc && \
+#    rm -rf /var/lib/apt/lists/*
+
```
you should install the tools you need as part of the build. I don't think it will make much difference in the build time or the overall size

robyww commented on docker/proxy/Dockerfile:
```diff
@@ -0,0 +1,17 @@
+
+EXPOSE 80 443
+
+ENTRYPOINT ["httpd-foreground"]
```
proxy should take parameters to to set which the backend it running so that it is not required to be on 8080.  They way we can start two proxy instances in front of two firelfy servers.

robyww commented on docker/proxy/proxyctl.sh:
```diff
@@ -0,0 +1,33 @@
+        docker pull ipac/proxy
+        docker run -d \
+                   -p 80:80 \
+                   -p 443:443 \
```
it you has more parameters to the docker containers the this file will have to deal with them

robyww commented on docker/proxy-dev/proxyctl.sh:
```diff
@@ -0,0 +1,36 @@
+        ;;
+  update)
+        cd /hydra/cm/firefly
+        gradle proxyDev:dockerImage
```
i don't have /hydra/cm/firefly. I think you should use the following>
```
DIR=$(dirname "${0}")
cd $DIR/../../../firefly
```

loitly commented:
I've just pushed a commit that address all of the issues identified in your review.  Please take a quick look and see if it addresses your concerns.  
You should be able to start one up out-of-the-box even if you do not have missing properties identified by tatiana.
Thanks.

```
./docker/proxy-dev/proxyctl.sh

Usage: proxyctl.sh [start|stop|shell|clean|update]
  optional parameters for update only:
    -http|--http_port <number>
    -https|--https_port <number>
    -host|--docker_host <x.x.x.x>  : ip address of the host running firefly
    -firefly|--firefly_port <number>

```
